hoomd/operation.py

```diff
@@ -178,8 +178,13 @@ def __del__(self):
 
     def detach(self):
         if self.is_attached:
+            # cache completion status for later use
+            self._is_complete = self.complete
+
+            # bring parameters back to Python
             self._unapply_typeparam_dict()
             self._update_param_dict()
+            self._cpp_obj.notifyDetach()
```

Comment:
Do we need this in the C++ code too? Thus, far the idea would be that dependencies would by handled in the Python layer and deal with the C++ objects accordingly.